rudder-agent/SOURCES/rudder.init

```diff
@@ -0,0 +1,219 @@
+#!/bin/sh
+
+#==============================================
+# Start/stop script for all Rudder components #
+#==============================================
+#
+### BEGIN INIT INFO
+# Provides:		rudder
+# Required-Start:	$local_fs $remote_fs $network $syslog
+# Required-Stop:	$local_fs $remote_fs $network $syslog
+# Default-Start:	2 3 4 5
+# Default-Stop:		0 1 6
+# Short-Description:	Rudder server
+# Description:		Rudder server including all components required to run
+### END INIT INFO
+#
+# Copyright (C) 2011-2015 Normation
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# GPL License: http://www.gnu.org/licenses/gpl.txt
+#
+#====================================================================
+
+# Variables
+
+## Colors configuration
+ECHO_BIN="/bin/echo -e"
+
+### Enable colors only if stdout is a terminal
+if [ -t 1 ]; then
+  GREEN="\\033[1;32m"
+  RED="\\033[1;31m"
+  BLUE="\\033[1;34m"
+  PINK="\\033[1;35m"
+  WHITE="\\033[0;02m"
+  WHITELIGHT="\\033[1;08m"
+  YELLOW="\\033[1;33m"
+  CYAN="\\033[1;36m"
+  NORMAL="\\033[0;39m"
+  COPYPASTE="${ECHO_BIN} ${GREEN}---8<---${NORMAL}"
+else
+  GREEN=""
+  RED=""
+  BLUE=""
+  PINK=""
+  WHITE=""
+  WHITELIGHT=""
+  YELLOW=""
+  CYAN=""
+  NORMAL=""
+  COPYPASTE="${ECHO_BIN} ---8<---"
+fi
+
+## Active roles
+
+HAVE_SLAPD=0
+HAVE_POSTGRESQL=0
+HAVE_JETTY=0
+HAVE_AGENT=1
+
+### Check slapd's status if we are rudder-inventory-ldap
+[ -e /opt/rudder/etc/server-roles.d/rudder-inventory-ldap ] && HAVE_SLAPD=1
+
+### Check PostgreSQL's status if we are rudder-db
+[ -e /opt/rudder/etc/server-roles.d/rudder-db ] && HAVE_POSTGRESQL=1
+
+### Check Jetty's status if we are rudder-webapp or rudder-inventory-endpoint
+[ -e /opt/rudder/etc/server-roles.d/rudder-inventory-endpoint ] && HAVE_JETTY=1
+[ -e /opt/rudder/etc/server-roles.d/rudder-webapp ] && HAVE_JETTY=1
+
+## Service names
+APACHE2=apache2
+if [ ! -x /etc/debian_version ] && [ ! -x /etc/SuSE-release ]; then
+  # We are not on Debian or SuSE, assuming a RHEL-like system
+  APACHE2=httpd
+fi
+RUDDER_SLAPD=rudder-slapd
+RUDDER_JETTY=rudder-jetty
+RUDDER_AGENT=rudder-agent
+POSTGRESQL=postgresql
+
+## PID files
+POSTGRESQL_PID_FILE=`find /var/lib -iname postmaster.pid`
+
+## Default return code
+RETURN=0
+
+# Functions
+
+service_action()
+{
+# Arguments:
+## Service name
+## Action to trigger (start/stop/...)
+## Is the role present on this machine (0/1)
+
+# If the role is present, run the action
+if [ "${3}" -eq 1 ]; then
+  service ${1} ${2} || RETURN=${?}
```

Comment:
As a precaution you should use "${var}" instead of ${var} when it's used a parameter